### PR TITLE
fix markdown request renderer

### LINKED ISSUE #14208: [Theia AI] HTML is not escaped in the Chat View
If you use HTML in the users message, it is not escaped, e.g. use

`Theia IDE Screenshot`

in the message. This will send "Theia IDE Screenshot" only to the underlying LLM.

### PR DESCRIPTION
**Pull request by eneufeld**
#### What it does

Currently requests in the ai chat view use a markdown renderer which parses html input and tries to render this. This leads to unexpected behavior when using html in the query.
By deactivating html support there, an alternative is needed as the markdown renderer just renders an empty paragraph if html is now passed to it. So to render the actual text, a span is created and filled with the text content.

fixes #14208
#### How to test

Use a string like `Theia IDE Screenshot` in you query.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx [hunk 1]
```diff
@@ -38,34 +38,51 @@ export class MarkdownPartRenderer implements ChatResponsePartRenderer<MarkdownCh
...
+        //     const html = this.markdownIt.render(response.content);
+        //     host.innerHTML = DOMPurify.sanitize(html, {
+        //         ALLOW_UNKNOWN_PROTOCOLS: true // DOMPurify usually strips non http(s) links from hrefs
+        //     });
+        //     while (ref?.current?.firstChild) {
+        //         ref.current.removeChild(ref.current.firstChild);
+        //     }
+
+        //     ref?.current?.appendChild(host);
+        // }, [response.content]);
```
Comment (sdirix) [inline packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx:56]: should be removed

File: packages/ai-chat-ui/src/browser/chat-tree-view/chat-view-tree-widget.tsx [hunk 2]
```diff
@@ -389,6 +377,27 @@ export class ChatViewTreeWidget extends TreeWidget {
...
+        const html = markdownIt.render(text);
+        host.innerHTML = DOMPurify.sanitize(html, {
+            ALLOW_UNKNOWN_PROTOCOLS: true // DOMPurify usually strips non http(s) links from hrefs
+        });
+        while (ref?.current?.firstChild) {
+            ref.current.removeChild(ref.current.firstChild);
+        }
+
+        ref?.current?.appendChild(host);
+    }, [node.request]);
```
Comment (sdirix) [inline packages/ai-chat-ui/src/browser/chat-tree-view/chat-view-tree-widget.tsx:396]: This is almost the same content as in the response renderer. I would like to suggest a custom hook `useMarkDownRendering(markDownText)` so we can reuse this logic in both.

The hook could then also be documented on why it exists and why it uses markdownit directly.

File: packages/ai-chat/src/common/chat-model.ts [hunk 3]
```diff
@@ -411,35 +410,35 @@ export class TextChatResponseContentImpl implements TextChatResponseContent {
 
 export class MarkdownChatResponseContentImpl implements MarkdownChatResponseContent {
     kind: 'markdownContent' = 'markdownContent';
-    protected _content: MarkdownStringImpl = new MarkdownStringImpl();
+    protected _content: string;
```
Comment (sdirix) [inline packages/ai-chat/src/common/chat-model.ts:413]: shouldn't we keep the markdown string content? depending on its attributes we could then hand over different options to markdownit. Thereby we could allow html embeddings if someone wants to have that.

File: packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx [hunk 4]
```diff
@@ -38,34 +38,53 @@ export class MarkdownPartRenderer implements ChatResponsePartRenderer<MarkdownCh
...
+
+        // return <div ref={ref}></div>;
+        return <MarkdownRender response={response} />;
     }
 
 }
 
-export const MarkdownWrapper = (props: { data: MarkdownString, renderCallback: (md: MarkdownString) => HTMLElement }) => {
+const MarkdownRender = ({ response }: { response: MarkdownChatResponseContent | InformationalChatResponseContent }) => {
     // eslint-disable-next-line no-null/no-null
```
Comment (sdirix) [inline packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx:56]:
```suggestion
```

File: packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx [hunk 5]
```diff
@@ -38,34 +38,53 @@ export class MarkdownPartRenderer implements ChatResponsePartRenderer<MarkdownCh
...
+ * overriden by theia with a monaco version. This monaco version strips all html
+ * tags from the markdown with empty content.
+ * This leads to unexpected behavior when rendering markdown with html tags.
+ *
+ * @param markdown the string to render as markdown
+ * @returns the ref to use in an element to render the markdown
+ */
+export const useMarkdownRendering = (markdown: string) => {
+    // eslint-disable-next-line no-null/no-null
+    const ref = useRef<HTMLDivElement | null>(null);
```
Comment (sdirix) [inline packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx:74]: Is there a benefit of using `null` over `undefined`?

File: packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx [hunk 6]
```diff
@@ -38,34 +38,53 @@ export class MarkdownPartRenderer implements ChatResponsePartRenderer<MarkdownCh
...
     render(response: MarkdownChatResponseContent | InformationalChatResponseContent): ReactNode {
         // TODO let the user configure whether they want to see informational content
         if (InformationalChatResponseContent.is(response)) {
             // null is valid in React
             // eslint-disable-next-line no-null/no-null
             return null;
         }
-        return <MarkdownWrapper data={response.content} renderCallback={this.renderMarkdown.bind(this)}></MarkdownWrapper>;
+
+        // return <div ref={ref}></div>;
```
Comment (sdirix) [inline packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx:49]:
```suggestion
```

File: packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx [hunk 7]
```diff
@@ -38,34 +38,53 @@ export class MarkdownPartRenderer implements ChatResponsePartRenderer<MarkdownCh
...
+ * @param markdown the string to render as markdown
+ * @returns the ref to use in an element to render the markdown
+ */
+export const useMarkdownRendering = (markdown: string) => {
+    // eslint-disable-next-line no-null/no-null
+    const ref = useRef<HTMLDivElement | null>(null);
+    useEffect(() => {
+        const markdownIt = markdownit();
+        const host = document.createElement('div');
+        const html = markdownIt.render(markdown);
```
Comment (sdirix) [inline packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx:78]:
```suggestion
        // TODO: accept MarkdownString objects and set options accordingly
        const html = markdownIt.render(markdown);
```

**Review by sdirix — COMMENTED** [review]
Works for me!

I tested requests like:
```
Hi <span> there </span>
```

And I executed prompts like:
```
Paste me an html template. Do not wrap in code block
```

These requests and answers were shown correctly :+1: 
---
I only have minor comments

File: packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx (hunk 5 above)
Comment (eneufeld) [reply packages/ai-chat-ui/src/browser/chat-response-renderer/markdown-part-renderer.tsx:73]: the ref paramerter wants null, can't be used with undefined

**Review by sdirix — APPROVED** [review]
Works great

**Pull request merged**